core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1459,20 +1479,82 @@ abstract class RDD[T: ClassTag](
       // children RDD partitions point to the correct parent partitions. In the future
       // we should revisit this consideration.
       RDDCheckpointData.synchronized {
-        checkpointData = Some(new RDDCheckpointData(this))
+        checkpointData = Some(new ReliableRDDCheckpointData(this))
       }
     }
   }
 
   /**
-   * Return whether this RDD has been checkpointed or not
+   * Mark this RDD for local checkpointing using Spark's existing caching layer.
+   *
+   * This method is for users who wish to truncate RDD lineages while skipping the expensive
+   * step of replicating the materialized data in a reliable distributed file system. This is
+   * useful for RDDs with long lineages that need to be truncated periodically (e.g. GraphX).
+   *
+   * Local checkpointing sacrifices fault-tolerance for performance. In particular,
+   * checkpointed data is written to ephemeral local storage (memory or disk) instead of
```

Comment:
...written to local storage of the executors...
(no point mentioning memory or disk, more confusing)